### PR TITLE
Custom logger signed

### PR DESCRIPTION
**Pull request by Retler**
### Description
This PR streamlines logging  and adds a wrapper arround the imported logger in order to log metadata at every call to the log.

### Parent Issue
Issue #95 

### Comments for the Reviewers
- Keep an eye out for architectural improvements
- Do we need to log more metadata?

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [x] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [x] I have updated/added relevant documentation (readme, doc comments, etc).
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/app/authstorage_middleware.go [hunk 1]
```diff
@@ -40,13 +40,13 @@ func (app *App) AuthStorageUnaryServerInterceptor() grpc.UnaryServerInterceptor
...
 		if err != nil {
-			log.Errorf("NewDBAuthStore failed: %v", err)
+			log.Error(ctx, "NewDBAuthStore failed", err)
 			return nil, status.Errorf(codes.Internal, "error encountered while connecting to auth storage")
 		}
 		defer func() {
 			err := authStorage.Rollback(ctx)
 			if err != nil {
-				log.Error(err)
+				log.Error(ctx, "Perfroming rollback", err)
```
Comment (philip-cc) [inline encryption-service/app/authstorage_middleware.go:49]:
```suggestion
				log.Error(ctx, "Performing rollback", err)
```

File: encryption-service/app/authstorage_middleware.go [hunk 2]
```diff
@@ -63,13 +63,13 @@ func (app *App) AuthStorageStreamingInterceptor() grpc.StreamServerInterceptor {
...
 		if err != nil {
-			log.Errorf("NewDBAuthStore failed: %v", err)
+			log.Error(ctx, "NewDBAuthStore failed", err)
 			return status.Errorf(codes.Internal, "error encountered while connecting to auth storage")
 		}
 		defer func() {
 			err := authStorage.Rollback(ctx)
 			if err != nil {
-				log.Error(err)
+				log.Error(ctx, "Perfroming rollback", err)
```
Comment (philip-cc) [inline encryption-service/app/authstorage_middleware.go:72]:
```suggestion
				log.Error(ctx, "Performing rollback", err)
```

File: encryption-service/contextkeys/contextkeys.go [hunk 3]
```diff
@@ -19,5 +19,7 @@ const (
 	RequestIDCtxKey GlobalContextKey = iota
 	UserIDCtxKey
 	MethodNameCtxKey
+	ObjectIDCtxKey
+	ErrorCtxKey
```
[the remark this answers is not in the document]
Comment (Retler) [reply encryption-service/contextkeys/contextkeys.go:23]: fixed

File: encryption-service/logger/logger.go [hunk 4]
```diff
@@ -17,53 +17,44 @@ import (
...
-		log.WithFields(log.Fields{
-			"UserID":    userID,
-			"Method":    info.FullMethod,
-			"Status":    "request",
-			"RequestID": requestID,
-		}).Info("Request start")
-
-		res, err := handler(ctx, req)
+func init() {
+	log.SetFormatter(&log.JSONFormatter{})
```
Comment (bruno-cc) [inline encryption-service/logger/logger.go:24]: maybe we should decide on the format pending of if a TTY is present

Comment (philip-cc) [reply encryption-service/logger/logger.go:33]: Might be a bit out of scope for now, but could be part of #74 ?

Comment (0xbb) [reply encryption-service/logger/logger.go:34]: Nothing urgent and I also have to check how it currently looks in a terminal.

Comment (0xbb) [reply encryption-service/logger/logger.go:34]: So feel free to close this conversation.

Comment (0xbb) [reply encryption-service/logger/logger.go:34]: Sure feel free to close this conversation.

**Review by philip-cc — APPROVED** [review]
I'm happy with this :)

**Pull request merged**